### ISSUE TITLE
CatalogResolver.resolveInputStream returns an InputStream, but should be a javax.xml.transform.Source derived result

### ISSUE THREAD

**Issue by GoogleCodeExporter** [sshyran/genxdm#70]
```
For a specific use case, a WSDL with a collection of inline schemas needs to 
resolved within the WSDL file. The easiest way to 

This should be accompanied by a test case for importing a WSDL with inline 
schemas that use forward, backward, and cyclic references across the schemas 
contained in that WSDL.

Ideally, we also extend the "Source" notion of "DOMSource" with a corresponding 
"GenXDMSource".
```

Original issue reported on code.google.com by `eric%tib...@gtempaccount.com` on 11 Oct 2011 at 9:44

**Comment by GoogleCodeExporter** [comment]
```
Perhaps the above should be turned into multiple bugs, but I wanted to put the 
use-case in one place so that we could reference it.
```

Original comment by `eric%tib...@gtempaccount.com` on 11 Oct 2011 at 9:44

**Comment by GoogleCodeExporter** [comment]
```
Do which what now?

First, a method called "resolveInputStream" clearly ought to return an input 
stream, not something else.  So if we wanted something other than an input 
stream from a catalog, we'd presumably want to ask to 
resolveSomeOtherAbstraction.

Second, why would we introduce javax.xml.transform abstractions into schema 
parsing?  CatalogResolver is not used (or intended for use) outside of schema 
parsing.  Not for validation, not for anything else; it's a parse-time 
abstraction intended to permit a URI to be turned into a parseable input stream 
with the (optional) assistance of a catalog.

If we need to accept something weird to deal with WSDL and WSDL-like dain 
bramage, we need to understand the use cases much more clearly.  I don't know 
that any schema parser author is going to care to be required to implement the 
bewildering bizarreness of WSDL schema abstractions, which effectively redefine 
schema concepts of visibility (meaning that a parser for schemas in WSDL has to 
(mis)behave rather differently than a Schema-specification-compliant schema 
parser).  I very much doubt that the way to deal with the web services oddities 
and deformations of schema is to introduce abstractions from 
javax.xml.transform into the parse process.

So ... since the current parser can't understand anything except an input 
stream, what's the genesis of the surprising desire to return a transformation 
input source rather than an input stream during schema uri catalog-assisted 
resolution?

A final note: I can see us implementing GenXDMSource (or XDMSource, even), but 
not as an extension of DOMSource.  We should, if we do it (which certainly 
ought to be a separate bug report!) (and I'm in favor), directly implement 
javax.xml.transform.Source.  That, however, doesn't seem at all related to 
schema parsing.

In short: do which what now?
```

Original comment by `aale...@gmail.com` on 12 Oct 2011 at 4:26

**Comment by GoogleCodeExporter** [comment]
```
Someone needs to use javax.xml.transform.stream.StreamSource 

I'd vote strongly for a GenXDM-related Source & Results -- I know I'll need them
```

Original comment by `anli.shu...@gmail.com` on 12 Oct 2011 at 5:04

**Comment by GoogleCodeExporter** [comment]
```
Which someone?  Not the parser.  Please see issue 71, if the goal is to wrap a 
GenXDM tree in a javax.xml.transform abstraction.  I think that the mention of 
CatalogResolver, a schema model interface, is not helping.

On issue 71, it would help if we had an example of a use case, so that we had a 
better idea of how much of the javax.xml.transform interfaces ought to be 
implemented.

On *this* issue, it would help if we had a description of where and why 
CatalogResolver is being used in a fashion that makes expansion of or changing 
the interface a part of the summary.  It sounds, bluntly, like a completely 
left-field bizarro solution to a problem that isn't described, which makes 
things extremely awkward.
```

Original comment by `aale...@gmail.com` on 12 Oct 2011 at 5:16

**Comment by GoogleCodeExporter** [comment]
```
From further discussion in email, this appears to be associated with a desire 
to allow the schema parser to handle WSDL variations on schema.

In my opinion, putting support for parsing the schemas inside or associated 
with WSDL into the schema parser is absolutely wrong.  A WSDL processor can 
handle the redefinition of schema visibility that WSDL 1.1, particularly in 
WS-I mode, mandates, but a schema parser that attempts to do so is forced to 
operate in multiple mutually-contradictory modes.  The issues of visibility 
mean that the component provider implementation available to the parser have to 
be broken in a particular fashion in order to enable resolution. Changing the 
catalog resolver (and especially changing its return type) won't do the job.

Now, it may be that we actually want to be able to resolve (using a catalog?) 
something during schema parsing that isn't an input stream.  I'm not convinced 
that this is a bug, so remarking it as an enhancement (but in my opinion, the 
enhancement necessary is a WSDL parser, not a superficial change to the 
resolution interfaces for the schema parser).
```

Original comment by `aale...@gmail.com` on 12 Oct 2011 at 6:46
- Added labels: **Component-Processors**, **Priority-Low**, **Type-Enhancement**
- Removed labels: **Priority-Medium**, **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
Whatever is done, using the execrably incomplete and nearly useless Source and 
Result interfaces seems less than likely to help. Marking as post-1.0.
```

Original comment by `aale...@gmail.com` on 26 Jul 2012 at 7:59
- Added labels: **Milestone-Future**

**Comment by GoogleCodeExporter** [comment]
```
Changed owner
```

Original comment by `eric%tib...@gtempaccount.com` on 6 May 2014 at 9:45
